File: WoB_Client_Basic/Assets/Network/Request/RequestLobby.cs
/*
ADDED REQUESTLOBBY CLASS 
CREATED FOR HW4 
This class is modified from the original WOB
to showcase understanding of WOB code.
Incorporated and defined new gamestatus protocols
May not be used in Infection modified by Tony and Ibrahim
 */
using UnityEngine;

using System;

public class RequestLobby : NetworkRequest {

	public RequestLobby() {
		request_id = Constants.CMSG_LOBBY;
	}
	
	public void send(int user_id, int size, string password) {
	    packet = new GamePacket(request_id);
		packet.addInt32(user_id);
		packet.addInt32(size);
		packet.addString(password);
	}
}

File: WoB_Server_Basic/src/networking/request/RequestLobby.java
/*
NEW LOBBY REQUEST
CREATED FOR HW4 
This class is modified from the original WOB
to showcase understanding of WOB code. May not 
be used in Infection created by Tony.
 */

package networking.request;

import networking.response.ResponseLobby;
import utility.Log;

import java.io.IOException;

public class RequestLobby extends GameRequest{

    private String host;
    private ResponseLobby responseLobby;

    public String getHost() {
        return host;
    }

    public void setHost(String host) {
        this.host = host;
    }

    @Override
    public void parse() throws IOException {
        responses.add(responseLobby = new ResponseLobby());
    }

    @Override
    public void doBusiness() throws Exception {
        Log.printf("Lobby request recieved");
    }
}